pkgs/by-name/cl/cloudquery/package.nix

```diff
@@ -0,0 +1,52 @@
+{
+  lib,
+  buildGoModule,
+  fetchFromGitHub,
+}:
+
+buildGoModule rec {
+  pname = "cloudquery";
+  version = "6.11.2";
+
+  src = fetchFromGitHub {
+    owner = "cloudquery";
+    repo = "cloudquery";
+    rev = "refs/tags/cli-v${version}";
+    hash = "sha256-zVzHPAN/UAVQF56qcoIFOqtX/Aek3+1lO2O965UYxPM=";
+  };
+
+  vendorHash = "sha256-9YvsVflHr4QSVCYCcWg1Okmncc1sNteMJEAdz92BIgI=";
+
+  CGO_ENABLED = 0;
+
+  modRoot = "cli";
+
+  # Some checks fail to download deps
+  doCheck = false;
+
+  ldflags = [
+    "-s"
+    "-w"
+    "-X github.com/cloudquery/cloudquery/cli/cmd.Version=${version}"
+  ];
+
+  postInstall = ''
+    mv $out/bin/cli $out/bin/cloudquery
+    rm $out/bin/gen
+  '';
```

Comment:
```suggestion
  '' + lib.optionalString (stdenv.buildPlatform.canExecute stdenv.hostPlatform) ''
    installShellCompletion --cmd cloudquery \
      --bash <($out/bin/cloudquery completion bash) \
      --fish <($out/bin/cloudquery completion fish) \
      --zsh <($out/bin/cloudquery completion zsh)
  '';
```

Reply:
I get build errors with the code above. 

Why do we need this? I built the original app from source and also downloaded the pre-built binary and neither include shell completion.

Reply:
> I get build errors with the code above.

Add `installShellFiles` to nativeBuildInputs.

> Why do we need this? I built the original app from source and also downloaded the pre-built binary and neither include shell completion.

https://github.com/cloudquery/cloudquery/blob/2f78b2b9d5c1b5438332443bbe6d4710c94f8a49/cli/.goreleaser.yaml#L96-L102

Reply:
> I get build errors with the code above.

Probably you didn't put `stdenv` at the top or didn't put `installShellFiles` into `nativeBuildInputs`. Here's the documentation for installing shell completions: https://github.com/NixOS/nixpkgs/blob/d18ef8212246b78339007d745738aa4ce5d34230/doc/hooks/installShellFiles.section.md#installshellcompletion-installshellfiles-installshellcompletion

<details><summary>Full diff (click here to expand)</summary>

```diff
diff --git a/pkgs/by-name/cl/cloudquery/package.nix b/pkgs/by-name/cl/cloudquery/package.nix
index 3b245cf82414..d1ff8af97edf 100644
--- a/pkgs/by-name/cl/cloudquery/package.nix
+++ b/pkgs/by-name/cl/cloudquery/package.nix
@@ -2,6 +2,8 @@
   lib,
   buildGoModule,
   fetchFromGitHub,
+  installShellFiles,
+  stdenvNoCC,
 }:
 
 buildGoModule rec {
@@ -21,6 +23,10 @@ buildGoModule rec {
 
   modRoot = "cli";
 
+  nativeBuildInputs = [
+    installShellFiles
+  ];
+
   # Some checks fail to download deps
   doCheck = false;
 
@@ -30,10 +36,17 @@ buildGoModule rec {
     "-X github.com/cloudquery/cloudquery/cli/cmd.Version=${version}"
   ];
 
-  postInstall = ''
-    mv $out/bin/cli $out/bin/cloudquery
-    rm $out/bin/gen
-  '';
+  postInstall =
+    ''
+      mv $out/bin/cli $out/bin/cloudquery
+      rm $out/bin/gen
+    ''
+    + lib.optionalString (stdenvNoCC.buildPlatform.canExecute stdenvNoCC.hostPlatform) ''
+      installShellCompletion --cmd cloudquery \
+        --bash <($out/bin/cloudquery completion bash) \
+        --fish <($out/bin/cloudquery completion fish) \
+        --zsh <($out/bin/cloudquery completion zsh)
+    '';
 
   meta = {
     description = "Data movement tool to sync data from any source to any destination";
```
</details>

> Why do we need this? I built the original app from source and also downloaded the pre-built binary and neither include shell completion.

Of course the package works without shell completions, it's just an extra nice to have thing that is more convenient on NixOS. If you want to have shell completions on other distros, you'd have to put a line in your .bashrc/.zshrc or whatever shell you're using for every single program.

Reply:
The shell completions work fine. Thanks!